pkg/asset/installconfig/aws/validation.go

```diff
@@ -381,6 +381,7 @@ func ValidateForProvisioning(client API, ic *types.InstallConfig, metadata *Meta
 		return nil
 	}
 
+	ctx := context.TODO()
```

Comment:
This is an empty context that never expires [1] so this is not doing anything for the API call. We want a `context.WithTimeout()` where we specify a reasonable timeout for the call to execute and return

[1] https://pkg.go.dev/context#TODO